pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_indexes.go

```diff
@@ -43,23 +38,20 @@ func IndexAPIExportsByAPIResourceSchemas(obj interface{}) ([]string, error) {
 }
 
 func IndexSyncTargetsByExports(obj interface{}) ([]string, error) {
-	synctarget, ok := obj.(*workloadv1alpha1.SyncTarget)
-	if !ok {
-		return []string{}, fmt.Errorf("obj is supposed to be a SyncTarget, but is %T", obj)
-	}
+	syncTarget := obj.(*workloadv1alpha1.SyncTarget)
 
-	clusterName := logicalcluster.From(synctarget)
-	if len(synctarget.Spec.SupportedAPIExports) == 0 {
+	clusterName := logicalcluster.From(syncTarget)
+	if len(syncTarget.Spec.SupportedAPIExports) == 0 {
 		return []string{client.ToClusterAwareKey(clusterName.Path(), reconcilerapiexport.TemporaryComputeServiceExportName)}, nil
 	}
 
-	keys := make([]string, 0, len(synctarget.Spec.SupportedAPIExports))
-	for _, export := range synctarget.Spec.SupportedAPIExports {
-		if len(export.Export) == 0 {
```

Comment:
FYI @davidfestal @sttts I'm pretty sure this is a bug, right?

Reply:
looks like, yes

Reply:
There is a _chance_ this is the root cause of #2762

Reply:
I agree it seems a bug that might be related to the mentioned issue.